test/hotspot/jtreg/serviceability/jvmti/ObjectMonitorUsage/libObjectMonitorUsage.cpp

```diff
@@ -225,32 +177,26 @@ Java_ObjectMonitorUsage_setTestedMonitor(JNIEnv *jni, jclass cls, jobject monito
     jni->DeleteGlobalRef(tested_monitor);
   }
   tested_monitor = (monitor != nullptr) ? jni->NewGlobalRef(monitor) : nullptr;
-  waits_to_enter = 0;
-  waits_to_be_notified = 0;
-
-  err = jvmti->SetEventNotificationMode(event_mode, JVMTI_EVENT_MONITOR_CONTENDED_ENTER, nullptr);
-  check_jvmti_status(jni, err, "setTestedMonitor: error in JVMTI SetEventNotificationMode #1");
-
-  err = jvmti->SetEventNotificationMode(event_mode, JVMTI_EVENT_MONITOR_CONTENDED_ENTERED, nullptr);
-  check_jvmti_status(jni, err, "setTestedMonitor: error in JVMTI SetEventNotificationMode #2");
-
-  err = jvmti->SetEventNotificationMode(event_mode, JVMTI_EVENT_MONITOR_WAIT, nullptr);
-  check_jvmti_status(jni, err, "setTestedMonitor: error in JVMTI SetEventNotificationMode #3");
-
-  err = jvmti->SetEventNotificationMode(event_mode, JVMTI_EVENT_MONITOR_WAITED, nullptr);
-  check_jvmti_status(jni, err, "setTestedMonitor: error in JVMTI SetEventNotificationMode #4");
 }
 
-JNIEXPORT jint JNICALL
-Java_ObjectMonitorUsage_waitsToEnter(JNIEnv *jni, jclass cls) {
+static void wait_for_state(JNIEnv *jni, jthread thread, jint exp_state) {
   RawMonitorLocker rml(jvmti, jni, event_lock);
```

Comment:
The event_lock name is misleading, there are no events anymore.
Also, I am not sure if this lock is needed at all. How it is used? 

I think that wait_for_state is a good candidate to be added in the library. With some additional doc about states.

Reply:
Good suggestions, thanks. Implemented now.